pkg/virt-controller/watch/vm_test.go

```diff
@@ -5335,6 +5335,64 @@ var _ = Describe("VirtualMachine", func() {
 					Expect(vmConditionController.HasCondition(vm, virtv1.VirtualMachineRestartRequired)).To(BeTrue())
 				})
 			})
+
+			Context("Affinity", func() {
+				It("should work", func() {
+					testutils.UpdateFakeKubeVirtClusterConfig(kvInformer, &v1.KubeVirt{
+						Spec: v1.KubeVirtSpec{
+							Configuration: v1.KubeVirtConfiguration{
+								VMRolloutStrategy: &liveUpdate,
+								DeveloperConfiguration: &v1.DeveloperConfiguration{
+									FeatureGates: []string{virtconfig.VMLiveUpdateFeaturesGate},
+								},
+							},
+						},
+					})
+
+					vm, vmi := DefaultVirtualMachine(true)
+					addVirtualMachine(vm)
+
+					affinity := k8sv1.Affinity{
+						PodAffinity: &k8sv1.PodAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: []k8sv1.PodAffinityTerm{
+								{
+									LabelSelector: &metav1.LabelSelector{
+										MatchLabels: map[string]string{
+											"test": "nmc",
+										},
+									},
+								},
+							},
+						},
+					}
+					vm.Spec.Template.Spec.Affinity = affinity.DeepCopy()
+					Expect(vmiInformer.GetIndexer().Add(vmi)).To(Not(HaveOccurred()))
+
+					vmiInterface.EXPECT().Patch(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(
+						func(_, _, _ any, data []byte, _ any, _ ...any) (*virtv1.VirtualMachineInstance, error) {
+							type P struct {
+								Op    string
+								Path  string
+								Value k8sv1.Affinity
+							}
+
+							patches := []P{}
+
+							Expect(json.Unmarshal(data, &patches)).To(Not(HaveOccurred()))
+							Expect(patches).To(HaveLen(1))
```

Comment:
Why don't we just use patchOperation?
```suggestion
							patches := []patch.PatchOperation{}

							Expect(json.Unmarshal(data, &patches)).To(Not(HaveOccurred()))
							Expect(patches).To(HaveLen(1))
							aff := patches[0].Value.(k8sv1.Affinity)
```

Reply:
This doesn't work well. The `Unmarshal` will produce a `map[string]interface{}` which would then need to be converted to `Affinity` and then we could use the `equality.Semantic.DeepEqual`.

Reply:
Can you please add a small code comment that explains that? I have also found it non-intuitive